### PR TITLE
FIP-0008: draft potential test cases.

### PR DESCRIPTION
**Pull request by raulk**
These would likely translate to test vectors.

### CODE REVIEW

**Comment by ZenGround0** [comment]
Great, I will plan to include these as actors generated test vectors once the framework is finished.

@anorth you can help by writing any of these as vm tests during implementation.

File: FIPS/fip-0008.md [hunk 1]
```diff
@@ -100,7 +100,18 @@ Thus, such changes must be delivered through a major version upgrade to the acto
...
+
+1. PreCommitSectorBatch with # of sectors 10, with all preconditions succeeding => OK.
+2. PreCommitSectorBatch with # of sectors 1, with all preconditions succeeding => OK and gas cost similar to PreCommitSector.
+3. PreCommitSectorBatch with # of sectors 33, with all preconditions succeeding => failure due to violation of maximum sector count (32).
+4. PreCommitSectorBatch with empty sector array, with all preconditions succeeding => failure?
+5. PreCommitSectorBatch with # of sectors 10, with insufficient (starting) balance to carry out pledge 1 => failure and state untouched.
+6. PreCommitSectorBatch with # of sectors 10, with insufficient balance to carry out pledge 10 => failure and state untouched.
+7. PreCommitSectorBatch with # of sectors 32, all sectors with deals => OK and deals activated.
+8. PreCommitSectorBatch with # of sectors 32, one deal in sector 20 => OK and deal activated.
+9. ...
```
Comment (jennijuju) [inline FIPS/fip-0008.md:114]:
```suggestion

```

**Review by jennijuju — APPROVED** [review]

File: FIPS/fip-0008.md [hunk 2]
```diff
@@ -100,7 +100,17 @@ Thus, such changes must be delivered through a major version upgrade to the acto
...
+
+Test cases will accompany implementation. Suggested scenarios include:
+
+1. PreCommitSectorBatch with # of sectors 10, with all preconditions succeeding => OK.
+2. PreCommitSectorBatch with # of sectors 1, with all preconditions succeeding => OK and gas cost similar to PreCommitSector.
+3. PreCommitSectorBatch with # of sectors 33, with all preconditions succeeding => failure due to violation of maximum sector count (32).
+4. PreCommitSectorBatch with empty sector array, with all preconditions succeeding => failure?
+5. PreCommitSectorBatch with # of sectors 10, with insufficient (starting) balance to carry out pledge 1 => failure and state untouched.
+6. PreCommitSectorBatch with # of sectors 10, with insufficient balance to carry out pledge 10 => failure and state untouched.
+7. PreCommitSectorBatch with # of sectors 32, all sectors with deals => OK and deals activated.
```
Comment (ZenGround0) [inline FIPS/fip-0008.md:112]: Deals aren't activated until ProveCommit so for this to make sense this test should include subsequent prove commit.

**Review by momack2 — APPROVED** [review]

**Review by anorth — APPROVED** [review]

File: FIPS/fip-0008.md (hunk 2 above)
Comment (jennijuju) [inline FIPS/fip-0008.md:112]:
```suggestion
7. PreCommitSectorBatch with # of sectors 32, all sectors with deals => OK and deals activated. (Note from @ZenGround0 Deals aren't activated until ProveCommit so for this to make sense this test should include subsequent prove commit.)
```

**Pull request merged**